lib/webtask.js

```diff
@@ -708,6 +756,66 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_auth_token(auth) {
+            var auth_token;
+            var decoded_auth_token;
+            var userinfo;
+            var aud_should_be;
+            var secret = auth.secret || ctx.config.default_auth0_client_secret.current;
+
+            if(auth.account)
+                aud_should_be = auth.account.client_id;
+            else
+                aud_should_be = req.headers.host + urlParse(req.url).pathname;
```

Comment:
Follwing our discussion about `aud` yesterday, at this level I think we should *require* that both `aud` and `secret` are specified in the webtask token.